bpf/bpf_lxc.c

```diff
@@ -1455,18 +1456,29 @@ ipv6_policy(struct __ctx_buff *ctx, int ifindex, __u32 src_label,
 	}
 
 #ifdef ENABLE_NODEPORT
-	if (ret == CT_NEW || ret == CT_REOPENED) {
-		bool dsr = false;
 # ifdef ENABLE_DSR
+	if (tuple->nexthdr == IPPROTO_TCP) {
+		if (ctx_load_bytes(ctx, l4_off + 12, &tcp_flags, 2) < 0)
+			return DROP_CT_INVALID_HDR;
+	}
+
+	if (ret == CT_NEW || ret == CT_REOPENED ||
+	    /* See comment in ipv4_policy() */
+	    (ret == CT_ESTABLISHED && (tuple->nexthdr != IPPROTO_TCP ||
+	    ((tcp_flags.value & TCP_FLAG_SYN) && !(tcp_flags.value & TCP_FLAG_ACK))))) {
```

Comment:
nit, can be simplified to single test in fast path: `tcp_flags.value & (TCP_FLAG_SYN|TCP_FLAG_ACK) == TCP_FLAG_SYN`